docs/sql/data_types/nested.md

```diff
@@ -58,12 +96,125 @@ SELECT {'birds':
             {'yes':'frog', 'maybe': 'salamander', 'huh': 'dragon', 'no':'toad'}
         };
 ```
+### Retrieving from Structs
+Retrieving a value from a struct can be accomplished using dot notation, bracket notation, or through [struct functions](../functions/nested#struct-functions) like `struct_extract`.
+```sql
+-- Use dot notation to retrieve the value at a key's location. This returns 1
+-- Note that we wrap the struct creation in parenthesis so that it happens first.
+-- This is only needed in our basic examples here, not when working with a struct column
+SELECT ({'x': 1, 'y': 2, 'z': 3}).x;
```

Comment:
I actually get an error with your example, and then making a slight tweak to it I also get the same error that I was seeing in my attempt in the prior comment. I can file an issue, unless I've made a mistake!

This fails with "Binder Error: Referenced table "a" not found!
Candidate tables: "tbl""
```sql
SELECT a.x FROM (VALUES ({'x': 1, 'y': 2, 'z': 3})) tbl(a);
```

This fails with "Not implemented Error: ColumnRef not implemented!"
```sql
SELECT tbl.a.x FROM (VALUES ({'x': 1, 'y': 2, 'z': 3})) tbl(a);
```

Reply:
You are probably running on the latest release instead of on the latest dev version, this was added [in this PR](https://github.com/duckdb/duckdb/pull/2578).

Reply:
You're exactly right! Got that working great on the latest Python --pre version. I'll implement it as this for now, but let me know if you want me to change it to the VALUES style!
```sql
SELECT a.x FROM (SELECT {'x':1, 'y':2, 'z':3} as a);
```